dask/dataframe/io/parquet/arrow.py

```diff
@@ -749,17 +960,48 @@ def read_partition(
 
         dfs = []
         for rg in row_group:
-            piece = pq.ParquetDatasetPiece(
-                path,
-                row_group=rg,
-                partition_keys=partition_keys,
-                open_file_func=partial(fs.open, mode="rb"),
-            )
-            arrow_table = cls._parquet_piece_as_arrow(
-                piece, columns, partitions, **kwargs
-            )
+            if pa_ds is not None and isinstance(rg, pa_ds.ParquetFileFragment):
+                # `rg` is already a `ParquetFileFragment`, pyarrow
+                # knows how to convert this to a `table`
+                cols = []
+                for name in columns:
+                    if name is None:
+                        if "__index_level_0__" in schema.names:
+                            columns.append("__index_level_0__")
```

Comment:
Do you remember what's the reason this is needed?

(I would think that the pandas metadata ensure we do this automatically)